cmd/kubeadm/app/phases/certs/pkiutil/pki_helpers.go

```diff
@@ -123,23 +180,22 @@ func TryLoadCertAndKeyFromDisk(pkiPath, name string) (*x509.Certificate, *rsa.Pr
 	case *ecdsa.PrivateKey:
```

Comment:
Oh, yeah. That's a limitation of `certutil` and the code for most of k8s.
We should eventually make it possible to generate ecdsa and other private keys, but right now rsa is hardcoded _in all places_. Given the little time we have left, you should just remove the statement and fail if the type is anything but rsa.

You can file an issue for k8s in general that the hardcoding on rsa should be removed.